--- unix/Makefile.in~	2009-01-24 16:07:12.000000000 +0100
+++ unix/Makefile.in	2009-01-24 16:12:56.000000000 +0100
@@ -143,14 +143,14 @@
 	ranlib $@
 
 Mk4tcl$(SHLIB_SUFFIX): mk4tcl.o mk4too.o $(LOBJS) $(LINK_SPECIAL_FILES)
-	$(SHLIB_LD) -o $@ mk4tcl.o mk4too.o $(LOBJS) $(LINK_SPECIAL_FLAGS)
+	$(SHLIB_LD) -o $@ mk4tcl.o mk4too.o $(LOBJS) $(LINK_SPECIAL_FLAGS) $(TCL_LIB)
 
 Mk4py$(LIB_SUFFIX): $(PYOBJS) $(LOBJS)
 	ar cru $@ $(PYOBJS) $(LOBJS)
 	ranlib $@
 
 Mk4py$(SHLIB_SUFFIX): $(PYOBJS) $(LOBJS) $(LINK_SPECIAL_FILES)
-	$(SHLIB_LD) -o $@ $(PYOBJS) $(LOBJS) $(LINK_SPECIAL_FLAGS)
+	$(SHLIB_LD) -o $@ $(PYOBJS) $(LOBJS) $(LINK_SPECIAL_FLAGS) $(PYTHON_LIB)
 
 Mk4lua$(SHLIB_SUFFIX): mk4lua.o $(LOBJS) $(LINK_SPECIAL_FILES)
 	$(SHLIB_LD) -o $@ mk4lua.o $(LOBJS) $(LINK_SPECIAL_FLAGS)
